### DISCUSSION TITLE
Passing data between map_request and map_response

### DISCUSSION THREAD

**Discussion by kt390256** [apollographql/router#4736] [General]
**Describe the solution you'd like**
Hi all, I am new to rust and the apollo router - a question I have is what's the best way here to pass data between map_requset and map_response?

Essentially I am trying to see how can I pass request id in the request stage to the response stage, so I can append the request id to the response header.

Thanks for reading the question and all efforts are appreciated!

**Comment by SimonSapin** [comment]
Hi! Assuming this is a Rust plugin, various request and response structs have [a `context` field](https://docs.rs/apollo-router/latest/apollo_router/struct.Context.html) with a key-value map of serializable data. The context is shared between various stages of the lifecycle of a given client request, so you could insert data in `map_request` and get it back in `map_response`

That same context is also [available from Rhai scripts](https://www.apollographql.com/docs/router/customizations/rhai-api/#requestcontext).

**Comment by BrynCooke** [comment]
I've converted this to a discussion as this is more of a general support request.

**Comment by kt390256** [comment]
Apology for the late response, I've tried the context object and it is working - thank you @SimonSapin and @BrynCooke 🙏
